loadstone/src/error.rs

```diff
@@ -7,7 +7,7 @@ use ufmt::{uwrite, uwriteln};
 /// Top level error type for the bootloader. Unlike the specific
 /// module errors, this error contains textual descriptions of the
 /// problem as it is meant to be directly reported through USART.
-#[derive(Debug, Copy, Clone, PartialEq, Format)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Format)]
```

Comment:
Why do you need `Eq`?

Reply:
its not strictly required, it just tells the compiler that the `Error` type has a proper equivalence relation instead of a partial equivalence relation.

the general advice is to derive it where possible, since it lets the compiler perform more aggressive optimizations